fieldservice_stage_validation/models/fsm_equipment.py

```diff
@@ -0,0 +1,22 @@
+# Copyright (C) 2020 Brian McMaster <brian@mcmpest.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from odoo import _, api, models
+from odoo.exceptions import ValidationError
+
+
+class FSMEquipment(models.Model):
+    _inherit = 'fsm.equipment'
+
+    @api.constrains('stage_id')
+    def _validate_stage_fields(self):
+        for rec in self:
+            stage = rec.stage_id
+            field_ids = stage.validate_field_ids
+            field_names = [x.name for x in field_ids]
+            values = rec.read(field_names)
+
+            for name in field_names:
+                if values[0][name] == False:
+                    raise ValidationError(_('Cannot move to stage "%s" ' \
+                                            'until the "%s" field is set.'
+                                            % (stage.name, name)))
```

Comment:
@brian10048 Does it work without the loop and something like:

```
if any(values[0][field_names]) == False:
    raise ValidationError()
```

Reply:
@max3903 I get an `unhashable type: 'list'` error when trying `values[0][field_names]`